src/model/channel/reaction.rs

```diff
@@ -397,11 +395,33 @@ impl From<EmojiIdentifier> for ReactionType {
     }
 }
 
-impl From<String> for ReactionType {
-    fn from(unicode: String) -> ReactionType { ReactionType::Unicode(unicode) }
+#[derive(Debug)]
+pub struct ReactionConversionError;
+
+impl Display for ReactionConversionError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "Failed to convert String to Reaction")
+    }
+}
+
+impl std::error::Error for ReactionConversionError {}
+
+impl TryFrom<String> for ReactionType {
+    type Error = ReactionConversionError;
+
+    fn try_from(emoji_string: String) -> std::result::Result<Self, Self::Error> {
+        if emoji_string.is_empty() {
+            return Err(ReactionConversionError)
+        }
+
+        if !emoji_string.starts_with('<') {
+            return Ok(ReactionType::Unicode(emoji_string))
+        }
```

Comment:
You're repeating the behaviour of the `TryFrom<&str>` impl redundantly.
```suggestion
```

Reply:
If the reaction is Unicode based, this avoids a string allocation